RecoTauTag/RecoTau/plugins/PATTauIDEmbedder.cc

```diff
@@ -0,0 +1,95 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/PatCandidates/interface/PATTauDiscriminator.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "FWCore/Utilities/interface/transform.h"
+
+class PATTauIDEmbedder : public edm::stream::EDProducer<>
+{
+public:
+
+	explicit PATTauIDEmbedder(const edm::ParameterSet&);
+	~PATTauIDEmbedder(){};
+
+	void produce(edm::Event&, const edm::EventSetup&);
+
+private:
+
+//--- configuration parameters
+	edm::EDGetTokenT<pat::TauCollection> src_;
+	typedef std::pair<std::string, edm::InputTag> NameTag;
+	std::vector<NameTag> tauIDSrcs_;
+	std::vector<edm::EDGetTokenT<pat::PATTauDiscriminator> > patTauIDTokens_;
+};
+
+PATTauIDEmbedder::PATTauIDEmbedder(const edm::ParameterSet& cfg)
+{
+	src_ = consumes<pat::TauCollection>(cfg.getParameter<edm::InputTag>("src"));
+	// it might be a single tau ID
+	if (cfg.existsAs<edm::InputTag>("tauIDSource")) {
+		tauIDSrcs_.push_back(NameTag("", cfg.getParameter<edm::InputTag>("tauIDSource")));
+	}
+	// or there might be many of them
+	if (cfg.existsAs<edm::ParameterSet>("tauIDSources")) {
```

Comment:
The difficulty here is that the tauID names are re-keyed to fixed names in the pat::Tau object. If I use VInputTag the modules in the InputTags might have any names and will not be found back when calling standard pat::Tau member functions. To circumvent this problem the same solution has been chosen by Alex here as for the standard pat::Tau (see: PATTauProducer.cc lines 74ff). I'm happy to agree on something better but not on a time scale shorter than a few weeks. Since the proposed solution is fine for the PATTauProducer (and has been ever since) might it be OK for this new module to do follow the same procedure for the time being? 

Cheers,
Roger